esp8266-fastled-webserver/Info.cpp

```diff
@@ -12,38 +12,110 @@ String WiFi_SSID(bool persistent) {
+  jsonDoc[F("wiFiChipId")]         = String(WIFI_getChipId(), HEX);     // HEX(uint32_t)
+  jsonDoc[F("flashChipId")]        = String(ESP.getFlashChipId(), HEX); // HEX(uint32_t)
+  jsonDoc[F("flashChipSize")]      = ESP.getFlashChipSize();            // uint32_t
+  jsonDoc[F("flashChipRealSize")]  = ESP.getFlashChipRealSize();        // uint32_t
+  jsonDoc[F("sdkVersion")]         = system_get_sdk_version();          // String; example: "2.2.2-"
+  jsonDoc[F("coreVersion")]        = ESP.getCoreVersion();              // String; example: ""
+  jsonDoc[F("bootVersion")]        = system_get_boot_version();         // uint8_t
+  jsonDoc[F("cpuFreqMHz")]         = ESP.getCpuFreqMHz();               // uint8_t
+  jsonDoc[F("freeHeap")]           = ESP.getFreeHeap();                 // uint32_t
+  jsonDoc[F("sketchSize")]         = ESP.getSketchSize();               // uint32_t
+  jsonDoc[F("freeSketchSpace")]    = ESP.getFreeSketchSpace();          // uint32_t
+  jsonDoc[F("resetReason")]        = ESP.getResetReason();              // longest: "Software/System restart"
+  jsonDoc[F("isConnected")]        = WiFi.isConnected();                // boolean
+  jsonDoc[F("wiFiSsidDefault")]    = WiFi_SSID(true);                   // SSID
+  jsonDoc[F("wiFiSSID")]           = WiFi_SSID(false);                  // SSID
+  jsonDoc[F("localIP")]            = WiFi.localIP().toString();         // IPAddress (IPv4 only currently)
+  jsonDoc[F("gatewayIP")]          = WiFi.gatewayIP().toString();       // IPAddress (IPv4 only currently)
+  jsonDoc[F("subnetMask")]         = WiFi.subnetMask().toString();      // IPAddress (IPv4 only currently)
+  jsonDoc[F("dnsIP")]              = WiFi.dnsIP().toString();           // IPAddress (IPv4 only currently)
+  jsonDoc[F("hostname")]           = WiFi.hostname();                   // hostname
+  jsonDoc[F("macAddress")]         = WiFi.macAddress();                 // macAddress
+  jsonDoc[F("autoConnect")]        = WiFi.getAutoConnect();             // boolean
+  jsonDoc[F("softAPSSID")]         = WiFi.softAPSSID();                 // SSID
+  jsonDoc[F("softAPIP")]           = WiFi.softAPIP().toString();        // IPv4
+  jsonDoc[F("BSSID")]              = WiFi.BSSIDstr();                   // macAddress
+  jsonDoc[F("softAPMacAddress")]   = WiFi.softAPmacAddress();           // macAddress
+
+  // what to do if overflow the ArduinoJSON buffer?
+  if (jsonDoc.overflowed()) {
+    // NOTE: this should *NEVER* occur if the maximum string lengths were calculated properly
+    return String(F("{}"));
+  };
 
-  return json;
+  // convert to String
+  String result;
+  // avoid heap fragmentation by pre-allocating 1023 bytes for result string.
+  // As of 2021-11-21, maximum 890 chars needed for minified result.
+  // As of 2021-11-21, maximum 1031 characters needed for prettified result.
+  result.reserve(1023); 
+  serializeJson(jsonDoc, result);
+  return result;
```

Comment:
Hi Tobi,

Yes, there is more that can be done!  More passing of `jsonDoc` references is likely to come.  These are smaller steps towards this.

for example, I was thinking of moving `jsonDoc` up, so that failures in `jsonDoc` creation / filling could return an error code.

This is an active area for me right now, so expect more in the near future.